### PR TITLE
Add HTTPS developer certificate management tool

### PR DESCRIPTION
**Pull request by javiercn**
Sample usages:
```
dotnet developercertificates http
dotnet developercertificates http -ep .\https.cer
dotnet developercertificates http -ep .\https.pfx -p aspnet
dotnet developercertificates http -ep .\https.pfx -p aspnet -t
```

### CODE REVIEW

**Comment by javiercn** [comment]
@glennc Check the samples and the messages

File: src/Microsoft.AspNetCore.DeveloperCertificates.Tools/Program.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+                app.OnExecute(() =>
+                    {
+                        app.ShowHelp();
+                        return Success;
+                    });
+
+                return app.Execute(args);
+            }
+            catch
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.DeveloperCertificates.Tools/Program.cs]: Log the error. Also, consider using `IReporter`.

Example: https://github.com/aspnet/DotNetTools/blob/c4788107f50d58bc730a420516f9f62505159652/src/Microsoft.Extensions.SecretManager.Tools/Program.cs#L39-L46

File: src/Microsoft.AspNetCore.DeveloperCertificates.Tools/Program.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    Console.WriteLine("There was an error saving the HTTPS developer certificate to the current user personal certificate store.");
+                    return ErrorSavingTheCertificate;
+                case EnsureCertificateResult.ErrorExportingTheCertificate:
+                    Console.WriteLine("There was an error exporting HTTPS developer certificate to a file.");
+                    return ErrorExportingTheCertificate;
+                case EnsureCertificateResult.FailedToTrustTheCertificate:
+                    Console.WriteLine("There was an error trusting HTTPS developer certificate.");
+                    return ErrorTrustingTheCertificate;
+                default:
+                    Console.WriteLine("Something went wrong. The HTTPS developer certificate could not be created.");
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.DeveloperCertificates.Tools/Program.cs]: Switch error output to `Console.Error.WriteLine`. Also, consider using `IReporter.Error`, like the other cli tools from this repo.

**Comment by javiercn** [comment]
@natemcmaster Updated

**Comment by natemcmaster** [comment]
Looks like you still need to update a few things to make build.cmd pass.
```
dotnet.exe : ERROR: Couldn't find package 'Microsoft.AspNetCore.CertificateGeneration.Task' in the repo
At C:\Users\appveyor\.dotnet\buildtools\korebuild\2.1.0-preview1-15523\scripts\common.psm1:7 char:5
+     & $cmd @args
+     ~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (ERROR: Couldn't...sk' in the repo:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
WARNING: Unnecessary exclusion in Microsoft.AspNetCore.CertificateGeneration.Task
Issue: BUILD_ITEMS_FRAMEWORK
Instance: *
WARNING: 0 error(s) and 1 warning(s) found with package Microsoft.AspNetCore.DeveloperCertificates.Tools (2.1.0-preview1-t000)
WARNING: DOTNETCLITOOL_PREFERCLIRUNTIME: DotnetCliTool package should contain a 'prefercliruntime' file
Showing JSON for ignore content:
{
  "Microsoft.AspNetCore.DeveloperCertificates.Tools": {
    "Exclusions": {
      "DOTNETCLITOOL_PREFERCLIRUNTIME": {
        "*": "Enter justification"
      }
    }
  }
}
ERROR: SUMMARY: 1 error(s) and 1 warning(s) found
```

**Pull request merged**